backend/src/test/java/ai/verta/modeldb/FindDatasetEntitiesTest.java

```diff
@@ -259,24 +265,33 @@ private static void createDatasetEntities() {
   private static void createDatasetVersionEntities() {
     DatasetVersionTest datasetVersionTest = new DatasetVersionTest();
 
-    long version = 1L;
-
+    GetBranchRequest getBranchRequest =
```

Comment:
this need to be redesigned so that the createDatasetVersion takes the responsibility of fetching the commit pointed by master and set it as parent_id
using
 
```
message GetBranchRequest {
    RepositoryIdentification repository_id = 1;
    string branch = 2;

    message Response {
        Commit commit = 1;
    }
}
```

this would also mean when we create the new version , we update the branch

Reply:
Added logic as per suggestions